eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -801,10 +825,16 @@ class Peer(val nodeParams: NodeParams,
     log.debug("got authenticated connection to address {}", connectionReady.address)
 
     if (connectionReady.outgoing) {
-      // we store the node address upon successful outgoing connection, so we can reconnect later
-      // any previous address is overwritten
-      nodeParams.db.peers.addOrUpdatePeer(remoteNodeId, connectionReady.address)
+      // We store the node address and features upon successful outgoing connection, so we can reconnect later.
+      // The previous address is overwritten: we don't need it since the current one works.
+      nodeParams.db.peers.addOrUpdatePeer(remoteNodeId, NodeInfo(connectionReady.remoteInit.features, Some(connectionReady.address)))
+    } else if (channels.nonEmpty) {
+      // If this is an incoming connection, we only store the peer details in our DB if we have channels with them.
+      // Otherwise nodes could DoS by simply connecting to us to force us to store data in our DB.
+      // We don't update the remote address, we don't know if we would successfully connect using the current one.
+      nodeParams.db.peers.addOrUpdatePeer(remoteNodeId, NodeInfo(connectionReady.remoteInit.features, None))
```

Comment:
Why do we need a write here? Isn't it enough to write when the channel is ready?